### PR TITLE
Remove jsonfile / cleanup installer path

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] General cleanup of installer.desktop.js
- [x] Defer loading installer.json until we know we need the data
- [x] More logging
- [x] Remove jsonfile dep

### CODE REVIEW

File: shared/desktop/app/installer.desktop.js [hunk 1]
```diff
@@ -1,59 +1,92 @@
...
 import * as SafeElectron from '../../util/safe-electron.desktop'
+import fs from 'fs'
+import path from 'path'
 import exec from './exec.desktop'
 import {keybaseBinPath} from './paths.desktop'
 import {quit} from './ctl.desktop'
-import {isLinux, isWindows} from '../../constants/platform'
+import {isDarwin} from '../../constants/platform'
 import logger from '../../logger'
-import UserData from './user-data.desktop'
```
Comment (chrisnojima) [inline shared/desktop/app/installer.desktop.js]: this class serves no real purpose. it just wraps some params that are passed in

File: shared/desktop/app/installer.desktop.js [hunk 2]
```diff
@@ -1,59 +1,92 @@
...
 
-import type {InstallResult} from '../../constants/types/rpc-gen'
+const file = path.join(SafeElectron.getApp().getPath('userData'), 'installer.json')
```
Comment (chrisnojima) [inline shared/desktop/app/installer.desktop.js]: in dev mode this path is ~/Library/Application Support/Electron (and not Keybase) due to some other issue. not worth fixing now but if you want to test this the installer.json file isn't in Keybase

File: shared/desktop/app/installer.desktop.js [hunk 3]
```diff
@@ -1,59 +1,92 @@
...
+    if (err && err.code === 'ENOENT') {
+      console.log('[Installer] loadHasPrompted: No installer.json file')
+    } else {
+      console.warn('[Installer] loadHasPrompted: Error loading state:', err)
+    }
+    return false
+  }
+}
 
-type State = {
```
Comment (chrisnojima) [inline shared/desktop/app/installer.desktop.js]: we just need to check if its false and set it to try and save it so we don't really keep this around at a top level

**Comment by chrisnojima** [comment]
@keybase/react-hackers 
cc: @cjb @songgao who've been in here

**Comment by zanderz** [comment]
windows LGTM, still no scary error on startup. FYI here is what it looks like for me - maybe we want to change the deprecated single instance stuff?

```
c:\work\src\github.com\keybase\client\shared\desktop\release\win32-x64\Keybase-win32-x64>
Loaded C:\Users\Steve\AppData\Local\Keybase\keybase.app.debug { enableActionLogging: false,
  enableStoreLogging: true,
  featureFlagsOverride: 'admin',
  forceImmediateLogging: false,
  filterActionLogs: false,
  forwardLogs: false,
  ignoreDisconnectOverlay: false,
  immediateStateLogging: false,
  printOutstandingRPCs: false,
  printRPC: true,
  reduxSagaLogger: false,
  reduxSagaLoggerMasked: false,
  showDevTools: true,
  skipSecondaryDevtools: false,
  userTimings: true }
KEYBASE_FEATURES is no longer supported edit the json file instead
(electron) 'app.makeSingleInstance(cb)' is deprecated. Use 'app.requestSingleInstanceLock() and app.on('second-instance', cb)' instead.
Version: 2.8.0-20181025083611+2abf1000cd
kb-service-check: starting keybase.exe
Not connected - starting keybase
```

**Comment by chrisnojima** [comment]
i did the deprecation in a different pr

**Comment by chrisnojima** [comment]
@zanderz is this approved?

**Pull request merged**